packages/rules-core/src/enforce-rationale.ts

```diff
@@ -0,0 +1,84 @@
+import { extname } from "path";
+import { PackageRule, FileContext, asBollLineNumber, Result, Success, Failure, Logger } from "@boll/core";
+
+export interface EnforceRationaleOptions {
+  [file: string]: string[];
+}
+
+const ruleName = "EnforceRationale";
+
+/**
+ * EnforceRationale ensures specified fields in package.json require a rationale for
+ * any additions.
+ */
+export class EnforceRationale implements PackageRule {
+  constructor(private logger: Logger, private options?: EnforceRationaleOptions) {}
+
+  get name(): string {
+    return ruleName;
+  }
+
+  async check(file: FileContext): Promise<Result[]> {
+    if (extname(file.filename) !== ".json") {
+      return [new Success(ruleName)];
+    }
+
+    if (!this.options) {
+      return [new Success(ruleName)];
+    }
+
+    const failures: Result[] = [];
+    const fileKey = file.filename.slice(process.cwd().length + 1);
```

Comment:
I feel like `filename.slice(process.cwd().length + 1)` should be a method on `file` called 'relativeFilename` or something similar.